MovingBlocks/DestinationSol
# NUI game menu

# Description
This pull request converts the in-game menu to NUI. It should be functionally identical to before.
### Before
![image](https://user-images.githubusercontent.com/24301287/141653175-2f34ed41-1974-4117-891a-94ba4f56369c.png)
### After
![image](https://user-images.githubusercontent.com/24301287/141653213-1b8200b7-e3ef-4b77-af4d-5218cdaf59e2.png)

# Testing
- Start a new game or continue an existing one
- Open the in-game menu with `Esc`
- Pressing the `Can sell used items` button should toggle its value between `Yes` and `No`
- Pressing the `Sound Volume` and `Music Volume` buttons should change the sound volume and music volume buttons respectively
- Pressing the `Respawn` button should respawn the player ship
- Pressing the `Exit` button should exit the game to the main menu
- Pressing the `Resume` button should close the in-game menu
- Pressing the `Esc` key should be the same as pressing the `Resume` button

# Notes
- This depends on #631 and #636.
- Until #631 and #636 are merged, it might be easier to view the changes made in this pull request by viewing the diff between this and #636 at https://github.com/BenjaminAmos/DestinationSol/compare/nui-inventory-screen...nui-game-menu.
